sirepo_bluesky/sirepo_detector.py

```diff
@@ -101,10 +124,22 @@ def update_parameters(self):
 
     def trigger(self):
         super().trigger()
-        datum_id = new_uid()
+
         date = datetime.datetime.now()
-        srw_file = Path('/tmp/data') / Path(date.strftime('%Y/%m/%d')) / \
-            Path('{}.dat'.format(datum_id))
+        file_name = new_uid()
+        self._resource_document, self._datum_factory, _ = compose_resource(
+            start={'uid': 'needed for compose_resource() but will be discarded'},
```

Comment:
Interesting. Maybe we can still generate a proper `uuid4` and put this note in a comment above?

Reply:
I am not sure I did this correctly. @tacaswell can you let me know the right way to name the data file?

Reply:
Start uid will get set by https://github.com/bluesky/bluesky/blob/566dce9e7ada95e2346c9aa61206c6a2f0a189bf/bluesky/bundlers.py#L435-L439 

We do not want to inject the current start document into the devices (so at this point in the code code can not (should not!) know the current start uid), however `compose_resource` from event_mode expects to get a start document and will extract a uid from (https://github.com/bluesky/event-model/blob/9d3651ef8a4d8e2c3f046524c92bd5ff0ceefa2d/event_model/__init__.py#L1604) 

So here we have to pass in a dict with the key `'uid'` so that we don't blow up _here_ which we expect to be overwritten _there_.

This probably should be fixed on the event-model side as well.